scripts/dev/curriculum/_commands/helpers.ts

```diff
@@ -0,0 +1,238 @@
+}
+
+type Comparison = {
+  slug: string;
+  screenshot: string;
+};
+export function renderComparison(
+  baseLabel: string,
+  targetLabel: string,
+  before: Comparison | null,
+  after: Comparison | null,
+) {
+  if (!before || !after) {
+    throw new Error("Missing");
+  }
+  const slug = before?.slug ?? after?.slug;
+
+  let imageHtml = `
+        <span id="${slug}" style="grid-column: 1 / -1"></span>
+        <div class="column-header" >
+            <span class="pill">${baseLabel}</span> ${slug}
+        </div>
+        <div class="column-header">
+            <span class="pill">${targetLabel}</span> ${slug}
+        </div>
+        <div class="column-header">
+            <span class="pill">diff</span> ${slug}
+        </div>
+    `;
+  if (before?.screenshot) {
+    imageHtml += `<div class="image-holder"><img src="${before.screenshot}" /></div>`;
+  } else {
+    imageHtml += `<div style="background: #eee; padding: 8px;">Missing</div>`;
+  }
+  if (after?.screenshot) {
+    imageHtml += `<div class="image-holder"><img src="${after.screenshot}" /></div>`;
+  } else {
+    imageHtml += `<div style="background: #eee; padding: 8px;">Missing</div>`;
+  }
+
+  if (before?.screenshot && after?.screenshot) {
+    imageHtml += `<div style="filter:invert(1)" class="image-holder">
+        <img src="${before.screenshot}" />
+        <div class="diff-image"><img src="${after.screenshot}" /></div>
+        </div>`;
+  } else {
+    imageHtml += `<div style="background: #eee; padding: 8px;">Missing</div>`;
+  }
+
+  return imageHtml;
+}
+
+export type Image = {
+  input: string;
+  top: number;
+  bottom: number;
+  left: number;
+};
+
+export async function combineScreenshots(
+  images: Image[],
+  outputFileName: string,
+  width: number,
+) {
+  if (images.length > 0) {
+    const totalHeight = images[images.length - 1]!.bottom;
+
+    await sharp({
+      create: {
+        width: width,
+        height: totalHeight,
+        channels: 4,
+        background: { r: 0, g: 0, b: 0, alpha: 0 },
+      },
+    })
+      .composite(images)
+      .toFile(outputFileName);
+  }
+}
+
+export async function screenshotPageCurrent(page: Page, path: string) {
+  const viewportHeight = await page.evaluate(() => {
+    return window.innerHeight;
+  });
+  const viewportWidth = await page.evaluate(() => {
+    return window.innerWidth;
+  });
+  const totalHeight = await page.evaluate(() => {
+    return document.body.scrollHeight;
+  });
+  const sections = Math.ceil(totalHeight / viewportHeight);
+
+  let imgHeight = 0;
+  const images: Image[] = [];
+  for (let i = 0; i < sections; i++) {
+    const sectionPath = `${path}_part${i + 1}.png`;
+
+    const def = {
+      input: sectionPath,
+      top: imgHeight,
+      bottom: Math.min(totalHeight, imgHeight + viewportHeight),
+      left: 0,
+    };
+
+    images.push(def);
+    imgHeight = def.bottom;
+  }
+
+  if (images.length < 1) {
+    throw new Error("Missing");
+  }
```

Comment:
Perhaps this error message could be a bit more descriptive?

Reply:
Fixed